src/network/type.hpp

```diff
@@ -19,7 +19,7 @@
 namespace FMM {
 namespace NETWORK{
 
-typedef int NodeID; /**< Node ID in the network, can be discontinuous int */
+typedef uint64_t NodeID; /**< Node ID in the network, can be discontinuous int */
```

Comment:
I will think more about this part.

Reply:
I needed to change this since OSM node IDs are 64-bit. At the same time, I understand this might introduce other issues (if other places still expect 32-bit integer or in format strings when writing output).

Alternatively, we could have an extra mapping from 64-bit to 32-bit node IDs (e.g. as an std::unordered_map), but that will also add complexity.

I think it would add convenience for users if the output could list the original OSM node IDs along the path, so that there is no extra relabeling step that needs to be considered.

Reply:
For this issue, if you check this matched information output https://fmm-wiki.github.io/docs/documentation/output/
I think one extra field should be designed there so you can easily export the node list. Then it becomes a problem whether you need to the start node ID or end node id of each edge.